bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/services/EvaluationService.java

```diff
@@ -140,7 +140,7 @@ private void contextEvaluate() {
 	}
 
 	protected void changeVariable(final String name, final Object value) {
-		if (name == null || variableFilter.contains(name)) {
```

Comment:
not sure why this variable filter exists (from Paul Webster back in 2011)... maybe to prevent collisions/overwrites in the wrapped IEclipseContext? in that case, maybe a child context would help?